js/src/logger.test.ts

```diff
@@ -19,6 +19,46 @@ configureNode();
 
 const { extractAttachments, deepCopyEvent } = _exportsForTestingOnly;
 
+test("verify MemoryBackgroundLogger intercepts logs", async () => {
+  // Log to memory for the tests.
+  _exportsForTestingOnly.simulateLoginForTests();
+
+  const memoryLogger = _exportsForTestingOnly.useTestBackgroundLogger();
+
+  const logger = initLogger({
+    projectName: "test",
+    projectId: "test-project-id",
+  });
+
+  await memoryLogger.flush();
+  expect(await memoryLogger.drain()).length(0);
+
+  // make some spans
+  const span = logger.startSpan({ name: "test-name-a" });
+  span.log({ metrics: { v: 1 } });
+  span.end();
+
+  const span2 = logger.startSpan({ name: "test-name-b" });
+  span2.log({ metrics: { v: 2 } });
+  span2.end();
+
+  await memoryLogger.flush();
+
+  const events = (await memoryLogger.drain()) as any[]; // FIXME[matt] what type should this be?
```

Comment:
If you don't care about the type, use `unknown`. Like `const events: unknown[] = await memoryLogger.drain()`.